docusaurus/docs/tutorials/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,235 @@
+      // Heads up! Only the following properties can be updated from the `configure()` function:
+      // - title
+      // - description
+      // - path
+      // - icon
+      // - category
+      return {
+        path: `/a/${pluginJson.id}/foo/timeseries`,
+      };
+    }
+
+    // Returning an empty object means no updates to any of the properties.
+    return {};
+  },
+});
+```
+
+### Open a modal from the `onClick()`
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+import { Button, Modal } from '@grafana/ui';
+
+export const plugin = new AppPlugin().addLink({
+  title: '...',
+  description: '...',
+  targets: [PluginExtensionPoints.DashboardPanelMenu],
+  // `event` - the `React.MouseEvent` from the click event
+  // `context` - the `context` object shared with the extensions
+  onClick: (event, { openModal, context }) =>
+    openModal({
+      title: 'My modal',
+      width: 500, // (Optional) - width of the modal in pixels
+      height: 500, // (Optional) - height of the modal in pixels
+
+      // Calling `onDismiss()` closes the modal
+      body: ({ onDismiss }) => (
+        <div>
+          <div>This is our modal.</div>
+
+          <Modal.ButtonRow>
+            <Button variant="secondary" fill="outline" onClick={onDismiss}>
+              Cancel
+            </Button>
+            <Button onClick={onDismiss}>Ok</Button>
+          </Modal.ButtonRow>
+        </div>
+      ),
+    }),
+});
+```
+
+## Components
+
+### Best practices for adding components
+
+- **Use the props** - check what props the extension point is passing to the components and use them to implement a more tailored experience
+- **Wrap your component with providers** - if you want to access any plugin specific state in your component make sure to wrap it with the necessary React context providers (e.g. for Redux)
+- **Use the enum for Grafana extension point ids** - if you are registering a component to one of the available Grafana extension points, make sure that you use the [`PluginExtensionPoints` enum exposed by the `@grafana/data`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L121) package.
+
+### Register a component
+
+```tsx
+import { PluginExtensionPoints } from '@grafana/data';
+
+export const plugin = new AppPlugin().addComponent({
+  title: 'User profile tab',
+  description: '...',
+  targets: [PluginExtensionPoints.UserProfileTab],
+  component: () => <div>This is a new tab on the user profile page.</div>,
+});
+```
+
+### Accessing plugin meta in a component
+
+You can use the `usePluginContext()` hook to access any plugin specific meta information inside your component. The hook [returns a `PluginMeta` object](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/plugin.ts#L62). (This can be useful because the component that you register from your plugin won't be rendered under the React tree of your plugin, but somewhere else in the UI.)
```

Comment:
```suggestion
You can use the `usePluginContext()` hook to access any plugin specific meta information inside your component. The hook returns a [`PluginMeta`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/plugin.ts#L62) object. This can be useful because the component that you register from your plugin won't be rendered under the React tree of your plugin, but somewhere else in the UI.
```